MessageProcessorModule/src/main/java/com/lotus/iot/mpm/datamodel/service/AssetStatusService.java

```diff
@@ -0,0 +1,53 @@
+package com.lotus.iot.mpm.datamodel.service;
+
+import com.lotus.iot.mpm.api.objectmodel.response.GeneralResponseCode;
+import com.lotus.iot.mpm.datamodel.service.objectmodel.Status;
+import com.lotus.iot.mpm.exception.OldMessageException;
+import com.lotus.iot.mpm.objectmodel.AssetStatusMessage;
+import com.lotus.iot.mpm.objectmodel.type.StatusType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Niloufar Mazloumpour
+ * @mail niloufar@mazloumpour.net
+ * @since 2018-05-22
+ */
+@Service
+public class AssetStatusService {
+    private AssetStatusMap assetStatusMap;
+
+    @Autowired
+    public AssetStatusService(AssetStatusMap assetStatusMap) {
+        this.assetStatusMap = assetStatusMap;
+    }
+
+    public StatusType updateStatus(AssetStatusMessage assetStatusMessage) throws OldMessageException {
+        String assetId = assetStatusMessage.getAssetId();
+        StatusType newStatusType = assetStatusMessage.getStatusType();
+        Date updatedAt = assetStatusMessage.getCreatedAt();
+        Status currentStatus = assetStatusMap.get(assetId);
+        //updateStatus asset status in map if it's a new status
+        if (currentStatus == null || updatedAt.getTime() >= currentStatus.getCreatedAt()) {
+            assetStatusMap.put(assetId, newStatusType, updatedAt);
```

Comment:
To simplify the `put()` method signature I'd suggest the following:
```suggestion
            assetStatusMap.put(assetId, new Status(assetStatusMessage));
```
To be able to do the above I would provide additional constructor for the `Status` class that could populate a `Status` instance from `AssetStatusMessage`:
```java
    public Status (AssetStatusMessage message) {
        this(message.statusType, message.createdAt);
    }
```